pytorch_lightning/utilities/data.py

```diff
@@ -406,16 +416,40 @@ def _replace_dataloader_init_method() -> Generator[None, None, None]:
         del subclass._old_init
 
 
-def _apply_fault_tolerant_automatic_capture_dataset_wrapper(dl_kwargs: Dict) -> Dict:
-    dataset = dl_kwargs["dataset"]
+def _apply_fault_tolerant_automatic_capture_dataset_wrapper(dataset: Dataset) -> Dataset:
     if isinstance(dataset, IterableDataset):
         # wrap the `IterableDataset` into a `CaptureIterableDataset` to record sampler states.
-        dl_kwargs["dataset"] = CaptureIterableDataset(dataset=dataset)
+        return CaptureIterableDataset(dataset=dataset)
     elif get_len(dataset) != float("inf"):
-        dl_kwargs["dataset"] = CaptureMapDataset(dataset=dataset)
+        return CaptureMapDataset(dataset=dataset)
     else:
```

Comment:
This else can now be removed

Reply:
Are you sure about it? It was there even before and we are still passing the same dataset to it

Reply:
I was referring to https://pheanex.github.io/pylint/#:~:text=Unnecessary%20else%20after%20return%20.,them%20containing%20a%20return%20statement.

Reply:
Oh, absolutely! I thought that it cannot happen anymore now! Thanks for the website, looks very useful. Wonder why precommit didn't catch it though